### PR TITLE
Replace Experiment schema with ExperimentRecipe

### LINKED ISSUE #149: Issues with the new experiment schema
While trying to get the first-run experiment to pass the new schema in this [patch](https://github.com/AllegroFox/messaging-system-inflight-assets/commit/7eef285c27925d8ae040256ba575a7d30c4f196a), we've run into the following observations/validation errors:

* The new schema doesn't allow `id`, `targeting`, and `filter_expresion` which are all necessary as:
  - `id` is used as the ID for each record in RS, also the validation script [assumes](https://github.com/mozilla/messaging-system-inflight-assets/blob/master/scripts/validate.py#L205) it's a required field for each item
  - `targeting` is used by x-man
  - `filter_expresion` is used for server-side filtering by RS
* The deletion of `arguments`, which is just a minor issue. We just need to update the validation script [here](https://github.com/mozilla/messaging-system-inflight-assets/blob/master/scripts/validate.py#L206) so it can check the recipe with the same schema
* How we populate those following required fields in the new schema:
  - startDate
  - endDate
  - proposedDuration
  - proposedEnrollment
  - referenceBranch

### PR DESCRIPTION
**Pull request by piatra**
Should fix #149

### CODE REVIEW

File: schema/messaging-experiments-82.schema.json [hunk 1]
```diff
@@ -1,7 +1,35 @@
...
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "A unique identifier for the Recipe"
+        },
+        "targeting": {
+          "type": "string",
+          "description": "JEXL expression using messaging system environment"
+        },
```
Comment (punamdahiya) [inline schema/messaging-experiments-82.schema.json:15]: Should add "filter_expression": {"type": "string",... similar to targeting

**Comment by punamdahiya** [comment]
"active",
        "features",
        "startDate",
        "endDate",
        "proposedDuration",
        "proposedEnrollment",
        "referenceBranch"


should not be required

**Comment by punamdahiya** [comment]
Throwing KeyError: 'groups' below when running with new schema, probably needs to be updated 

https://github.com/mozilla/messaging-system-inflight-assets/blob/master/scripts/validate.py#L142

**Comment by ncloudioj** [comment]
@punamdahiya #151 should fix the `groups` key error.

Chatted about the `filter_expression` with Andrei, and we agreed that this property was causing lots of confusion for the old schema, we'd like to deprecate it in the new one. @punamdahiya As a result, let's append those filters in `filter_experssion` to `targeting`.

For the other new required properties, I think we can just provide as many as we could, and use dummy values if they're not applicable to the experiment.

**Comment by ncloudioj** [comment]
@punamdahiya @piatra Hey folks, I've merged #151, let's merge this PR as well so that Punam and Emily can proceed with their test. Thanks!

**Review by ncloudioj — APPROVED** [review]
LGTM.

As for those new required fields about the experiment, let's provide some dummy values for now in the recipe assuming that doing so wouldn't cause any expected behavior in Fx.

**Comment by punamdahiya** [comment]
> LGTM.
> 
> As for those new required fields about the experiment, let's provide some dummy values for now in the recipe assuming that doing so wouldn't cause any expected behavior in Fx.

Sounds good, if there are recommended default values will be good to know. For now using below as default for  "aboutwelcome" recipe passes new schema

    active: true
    features: []
    startDate: ""
    endDate: ""
    proposedDuration: 0
    proposedEnrollment: 0
    referenceBranch: ""

**Review by punamdahiya — APPROVED** [review]

**Pull request merged**